src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -1461,4 +1467,31 @@ private boolean onSpongeLightningDamage(net.minecraft.entity.Entity entity, Dama
         }
     }
 
+    @Inject(method = "setDead", at = @At(value = "RETURN"))
+    private void onSetDead(CallbackInfo ci) {
+        if (ShouldFire.DESTRUCT_ENTITY_EVENT && !this.world.isRemote && !(EntityUtil.toNative(this) instanceof EntityLiving)) {
+            MessageChannel originalChannel = MessageChannel.TO_NONE;
+            try (final CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {
+                frame.pushCause(this);
```

Comment:
Pushing `this` as the cause of death seems useless. The event already has the entity obviously, and the entity itself in is no way 'causing' its own death here.

Reply:
True

Reply:
Wait it’s been modified since mine, okay